src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -368,27 +369,45 @@ public void forceRecipeRecheck() {
      */
     protected void trySearchNewRecipe() {
         long maxVoltage = getMaxVoltage();
-        Recipe currentRecipe;
+        Recipe currentRecipe = null;
+        Iterator<Recipe> recipeIterator;
         IItemHandlerModifiable importInventory = getInputInventory();
         IMultipleTankHandler importFluids = getInputTank();
 
         // see if the last recipe we used still works
         if (checkPreviousRecipe()) {
             currentRecipe = this.previousRecipe;
             // If there is no active recipe, then we need to find one.
-        } else {
-            currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
         }
-        // If a recipe was found, then inputs were valid. Cache found recipe.
-        if (currentRecipe != null) {
+        recipeIterator = getRecipeIterator(maxVoltage, importInventory, importFluids);
```

Comment:
Should move this after the current recipe check since there's no need to create the iterator if the current recipe will succeed

Reply:
fixed